pkg/image/association.go

```diff
@@ -35,8 +35,18 @@ func (e *ErrInvalidComponent) Error() string {
 	return fmt.Sprintf("image %q has invalid component %q", e.image, e.tag)
 }
 
-// Associations is a set of image Associations.
-type Associations map[string]Association
+// AssociationSet is a mapped for Association
+// searching
+type AssociationSet map[string]Association
+
+// Associations is a set of image Associations
+// mapped to their images
+type Associations map[string]AssociationSet
```

Comment:
Wdyt about
```suggestion
type AssociationsSet map[string]AssociationSet
```
with the above change